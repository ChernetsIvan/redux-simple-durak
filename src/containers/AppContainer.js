import { connect } from 'react-redux';

import App from './../components/App';

const mapStateToProps = store => ({
    isRenderSettingsForStartNewGame: store.game.isRenderSettingsForStartNewGame
});

export default connect(mapStateToProps)(App);