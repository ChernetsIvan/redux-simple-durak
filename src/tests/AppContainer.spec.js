import React from "react";
import renderer from "react-test-renderer";

import { Provider } from "react-redux";
import { createStore } from "redux";

import AppContainer from "./../containers/AppContainer";
import reducer from "./../reducers";

describe("AppContainer snapshot testing", () => {
  let store;

  beforeEach(() => {
    store = createStore(reducer);
  });

  test("Store in initial-state mode", () => {
    const component = renderer.create(
      <Provider store={store}>
        <AppContainer />
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("Dispatching with START_GAME_CLICKED-action", () => {
    store.dispatch({
      type: "START_GAME_CLICKED"
    });
    const component = renderer.create(
      <Provider store={store}>
        <GameScreenContainer />
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("Dispatching with BEGIN_GAME_CLICKED-action", () => {
    store.dispatch({
      type: "BEGIN_GAME_CLICKED"
    });
    const component = renderer.create(
      <Provider store={store}>
        <GameScreenContainer />
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
