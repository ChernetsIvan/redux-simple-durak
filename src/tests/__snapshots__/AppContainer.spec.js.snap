// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer snapshot testing Store in initial-state mode 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <div
      className="row justify-content-center"
    >
      <div
        className="col-auto"
      >
        <h1
          className="display-2"
        >
          Карточная игра
        </h1>
      </div>
    </div>
    <div
      className="row justify-content-center mt-3"
    >
      <div
        className="col-auto"
      >
        <h1
          className="display-3"
        >
          Простой дурак
        </h1>
      </div>
    </div>
    <div
      className="row justify-content-center mt-5"
    >
      <div
        className="col-auto"
      >
        <span
          className="mr-4"
        >
          <b>
            Первым ходит:
          </b>
        </span>
        <span
          className="mr-2"
        >
          <input
            checked={true}
            className=""
            name="PlayerFirst"
            onChange={[Function]}
            type="radio"
          />
          <span
            className="ml-2"
          >
            Игрок
          </span>
        </span>
        <span
          className="mr-2"
        >
          <input
            checked={false}
            name="AiFirst"
            onChange={[Function]}
            type="radio"
          />
          <span
            className="ml-2"
          >
            Компьютер
          </span>
        </span>
      </div>
    </div>
    <div
      className="row justify-content-center mt-3"
    >
      <div
        className="col-auto"
      >
        <button
          className="btn btn-secondary mr-3"
          onClick={[Function]}
        >
          <span>
            —
          </span>
           Поехали!
        </button>
      </div>
    </div>
    <div
      className="row justify-content-center mt-3"
    >
      <div
        className="col-auto"
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          Статистика игр
        </a>
      </div>
    </div>
  </div>
</div>
`;
